libs/sdk-ui-kit/src/Dialog/ShareDialog/ShareDialogBase/useShareDialogBase.ts

```diff
@@ -0,0 +1,199 @@
+        (grantee: GranteeItem) => {
+            setGranteesToAdd((state) => state.filter((g) => !areObjRefsEqual(g.id, grantee.id)));
+        },
+        [setGranteesToAdd],
+    );
+
+    const onGranteeAdd = useCallback(
+        (grantee: GranteeItem) => {
+            setGranteesToAdd((state) => [...state, grantee]);
+        },
+        [setGranteesToAdd],
+    );
+
+    const onAddGranteeButtonClick = useCallback(() => {
+        setDialogMode("AddGrantee");
+    }, [setDialogMode]);
+
+    const onAddGranteeBackClick = useCallback(() => {
+        setDialogMode("ShareGrantee");
+        setGranteesToAdd([]);
+    }, [setDialogMode, setGranteesToAdd]);
+
+    const onLoadGrantees = useCallback(
+        (grantees: GranteeItem[], groupAll: IGranteeGroupAll | undefined) => {
+            if (groupAll) {
+                setGrantees([...grantees, groupAll]);
+            } else {
+                setGrantees(grantees);
+            }
+
+            setIsGranteesLoading(false);
+        },
+        [setGrantees, setIsGranteesLoading],
+    );
+
+    return {
+        dialogMode,
+        isGranteesLoading,
+        grantees,
+        granteesToAdd,
+        granteesToDelete,
+        onLoadGrantees,
+        onSharedGranteeDelete,
+        onAddedGranteeDelete,
+        onGranteeAdd,
+        onAddGranteeButtonClick,
+        onAddGranteeBackClick,
+    };
+};
+
+/**
+ * @internal
+ */
+export interface IUseShareDialogBaseReturnType {
+    onAddedGranteeDelete: (grantee: GranteeItem) => void;
+    onSharedGranteeDelete: (grantee: GranteeItem) => void;
+    onAddGranteeBackClick: () => void;
+    onAddGranteeButtonClick: () => void;
+    onGranteeAdd: (grantee: GranteeItem) => void;
+    onSubmitShareGrantee: () => void;
+    onSubmitAddGrantee: () => void;
+    isGranteesLoading: boolean;
+    granteesToAdd: GranteeItem[];
+    dialogMode: DialogModeType;
+    isShareDialogDirty: boolean;
+    isAddDialogDirty: boolean;
+    sharedGrantees: GranteeItem[];
+    appliedGranteesWithOwner: GranteeItem[];
+}
+
+/**
+ * @internal
+ */
+export const useShareDialogBase = (props: IShareDialogBaseProps): IUseShareDialogBaseReturnType => {
+    const { shareStatus, sharedObjectRef, owner, onSubmit, onError } = props;
+    const {
+        dialogMode,
+        isGranteesLoading,
+        grantees,
+        granteesToAdd,
+        granteesToDelete,
+        onLoadGrantees,
+        onSharedGranteeDelete,
+        onAddedGranteeDelete,
+        onGranteeAdd,
+        onAddGranteeButtonClick,
+        onAddGranteeBackClick,
+    } = useShareDialogState();
+
+    const onLoadGranteesSuccess = useCallback(
+        (result: GranteeItem[]) => {
+            const groupAll = mapShareStatusToGroupAll(shareStatus);
+            onLoadGrantees(result, groupAll);
+        },
+        [onLoadGrantees, shareStatus],
+    );
+
+    useGetAccessList({ sharedObjectRef, onSuccess: onLoadGranteesSuccess, onError });
+
+    const isShareDialogDirty = useMemo(() => {
```

Comment:
Is this memo and the following one really necessary?
I think 
```ts
const isShareDialogDirty = granteesToDelete.length !== 0;
```
is much clearer.